[0.10.0-rc1] Bringing up a node as a meta only node causes panic

While bringing up a node as a meta only node

The config
```
[meta]
  enabled = true
  bind-address = "10.230.211.53:8088"
  http-bind-address = "10.230.211.53:8091"
...

[data]
  enabled = false
...

[http]
  enabled = true
  bind-address = "10.230.211.53:8086"
...
```

induces the following panic:

```
2016/01/29 19:22:22 Go version go1.4.3, GOMAXPROCS set to 16
2016/01/29 19:22:22 Using configuration at: /etc/influxdb/influxdb.conf
[meta] 2016/01/29 19:22:22 Starting meta service
[meta] 2016/01/29 19:22:22 Listening on HTTP: 10.230.211.53:8091
[metastore] 2016/01/29 19:22:22 Using data dir: /var/lib/influxdb/meta
[metastore] 2016/01/29 19:22:22 Node at 10.230.211.53:8088 [Follower]
[metastore] 2016/01/29 19:22:24 Node at 10.230.211.53:8088 [Leader]. peers=[10.230.211.53:8088]
[run] 2016/01/29 19:22:25 Listening for signals
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xb code=0x1 addr=0x0 pc=0x55f1d5]

goroutine 17 [running]:
github.com/influxdb/influxdb/services/meta.(*Client).data(0x0, 0x0)
	/tmp/tmp.WXyOm7f2Na/src/github.com/influxdb/influxdb/services/meta/client.go:215 +0x95
github.com/influxdb/influxdb/services/meta.(*Client).Databases(0x0, 0x0, 0x0, 0x0, 0x0, 0x0)
	/tmp/tmp.WXyOm7f2Na/src/github.com/influxdb/influxdb/services/meta/client.go:313 +0x55
github.com/influxdb/influxdb/cmd/influxd/run.(*Server).reportServer(0xc208082180)
	/tmp/tmp.WXyOm7f2Na/src/github.com/influxdb/influxdb/cmd/influxd/run/server.go:541 +0x3a
github.com/influxdb/influxdb/cmd/influxd/run.(*Server).startServerReporting(0xc208082180)
	/tmp/tmp.WXyOm7f2Na/src/github.com/influxdb/influxdb/cmd/influxd/run/server.go:534 +0x5e
created by github.com/influxdb/influxdb/cmd/influxd/run.func·003
	/tmp/tmp.WXyOm7f2Na/src/github.com/influxdb/influxdb/cmd/influxd/run/server.go:465 +0x15de

goroutine 1 [chan receive]:
main.(*Main).Run(0xc208039700, 0xc20800a010, 0x4, 0x4, 0x0, 0x0)
	/tmp/tmp.WXyOm7f2Na/src/github.com/influxdb/influxdb/cmd/influxd/main.go:96 +0x7a1
main.main()
	/tmp/tmp.WXyOm7f2Na/src/github.com/influxdb/influxdb/cmd/influxd/main.go:46 +0xdc

goroutine 6 [syscall]:
os/signal.loop()
	/root/.gvm/gos/go1.4.3/src/os/signal/signal_unix.go:21 +0x1f
created by os/signal.init·1
	/root/.gvm/gos/go1.4.3/src/os/signal/signal_unix.go:27 +0x35

goroutine 8 [IO wait]:
net.(*pollDesc).Wait(0xc20800c760, 0x72, 0x0, 0x0)
	/root/.gvm/gos/go1.4.3/src/net/fd_poll_runtime.go:84 +0x47
net.(*pollDesc).WaitRead(0xc20800c760, 0x0, 0x0)
	/root/.gvm/gos/go1.4.3/src/net/fd_poll_runtime.go:89 +0x43
net.(*netFD).accept(0xc20800c700, 0x0, 0x7f81bbb7aee8, 0xc208102000)
	/root/.gvm/gos/go1.4.3/src/net/fd_unix.go:419 +0x40b
net.(*TCPListener).AcceptTCP(0xc20802a9e8, 0x0, 0x0, 0x0)
	/root/.gvm/gos/go1.4.3/src/net/tcpsock_posix.go:234 +0x4e
net.(*TCPListener).Accept(0xc20802a9e8, 0x0, 0x0, 0x0, 0x0)
	/root/.gvm/gos/go1.4.3/src/net/tcpsock_posix.go:244 +0x4c
github.com/influxdb/influxdb/tcp.(*Mux).Serve(0xc20805a5a0, 0x7f81bbb81758, 0xc20802a9e8, 0x0, 0x0)
	/tmp/tmp.WXyOm7f2Na/src/github.com/influxdb/influxdb/tcp/mux.go:52 +0xc7
created by github.com/influxdb/influxdb/cmd/influxd/run.func·003
	/tmp/tmp.WXyOm7f2Na/src/github.com/influxdb/influxdb/cmd/influxd/run/server.go:379 +0x26f

goroutine 9 [chan receive]:
github.com/influxdb/influxdb/tcp.(*listener).Accept(0xc20807f190, 0x0, 0x0, 0x0, 0x0)
	/tmp/tmp.WXyOm7f2Na/src/github.com/influxdb/influxdb/tcp/mux.go:135 +0x80
github.com/influxdb/influxdb/services/meta.(*raftLayer).Accept(0xc20807ced0, 0x0, 0x0, 0x0, 0x0)
	/tmp/tmp.WXyOm7f2Na/src/github.com/influxdb/influxdb/services/meta/raft_state.go:325 +0x65
github.com/hashicorp/raft.(*NetworkTransport).listen(0xc20800c7e0)
	/tmp/tmp.WXyOm7f2Na/src/github.com/hashicorp/raft/net_transport.go:362 +0x50
created by github.com/hashicorp/raft.NewNetworkTransportWithLogger
	/tmp/tmp.WXyOm7f2Na/src/github.com/hashicorp/raft/net_transport.go:154 +0x2c4

goroutine 10 [select]:
github.com/hashicorp/raft.(*Raft).leaderLoop(0xc2080c2380)
	/tmp/tmp.WXyOm7f2Na/src/github.com/hashicorp/raft/raft.go:874 +0xe30
github.com/hashicorp/raft.(*Raft).runLeader(0xc2080c2380)
	/tmp/tmp.WXyOm7f2Na/src/github.com/hashicorp/raft/raft.go:838 +0x876
github.com/hashicorp/raft.(*Raft).run(0xc2080c2380)
	/tmp/tmp.WXyOm7f2Na/src/github.com/hashicorp/raft/raft.go:602 +0xc9
github.com/hashicorp/raft.*Raft.(github.com/hashicorp/raft.run)·fm()
	/tmp/tmp.WXyOm7f2Na/src/github.com/hashicorp/raft/raft.go:252 +0x27
github.com/hashicorp/raft.func·011()
	/tmp/tmp.WXyOm7f2Na/src/github.com/hashicorp/raft/state.go:152 +0x51
created by github.com/hashicorp/raft.(*raftState).goFunc
	/tmp/tmp.WXyOm7f2Na/src/github.com/hashicorp/raft/state.go:153 +0xe3

goroutine 11 [select]:
github.com/hashicorp/raft.(*Raft).runFSM(0xc2080c2380)
	/tmp/tmp.WXyOm7f2Na/src/github.com/hashicorp/raft/raft.go:509 +0x1043
github.com/hashicorp/raft.*Raft.(github.com/hashicorp/raft.runFSM)·fm()
	/tmp/tmp.WXyOm7f2Na/src/github.com/hashicorp/raft/raft.go:253 +0x27
github.com/hashicorp/raft.func·011()
	/tmp/tmp.WXyOm7f2Na/src/github.com/hashicorp/raft/state.go:152 +0x51
created by github.com/hashicorp/raft.(*raftState).goFunc
	/tmp/tmp.WXyOm7f2Na/src/github.com/hashicorp/raft/state.go:153 +0xe3

goroutine 12 [select]:
github.com/hashicorp/raft.(*Raft).runSnapshots(0xc2080c2380)
	/tmp/tmp.WXyOm7f2Na/src/github.com/hashicorp/raft/raft.go:1705 +0x466
github.com/hashicorp/raft.*Raft.(github.com/hashicorp/raft.runSnapshots)·fm()
	/tmp/tmp.WXyOm7f2Na/src/github.com/hashicorp/raft/raft.go:254 +0x27
github.com/hashicorp/raft.func·011()
	/tmp/tmp.WXyOm7f2Na/src/github.com/hashicorp/raft/state.go:152 +0x51
created by github.com/hashicorp/raft.(*raftState).goFunc
	/tmp/tmp.WXyOm7f2Na/src/github.com/hashicorp/raft/state.go:153 +0xe3

goroutine 13 [select]:
github.com/influxdb/influxdb/services/meta.(*raftState).logLeaderChanges(0xc2080622d0)
	/tmp/tmp.WXyOm7f2Na/src/github.com/influxdb/influxdb/services/meta/raft_state.go:145 +0x459
created by github.com/influxdb/influxdb/services/meta.(*raftState).open
	/tmp/tmp.WXyOm7f2Na/src/github.com/influxdb/influxdb/services/meta/raft_state.go:135 +0x12b2

goroutine 15 [runnable]:
net.accept(0x6, 0x0, 0x0, 0x0, 0x0, 0x0)
	/root/.gvm/gos/go1.4.3/src/net/sock_cloexec.go:55 +0xe9
net.(*netFD).accept(0xc20800c770, 0x0, 0x0, 0x0)
	/root/.gvm/gos/go1.4.3/src/net/fd_unix.go:416 +0x309
net.(*TCPListener).AcceptTCP(0xc20802a9f0, 0xc20811a380, 0x0, 0x0)
	/root/.gvm/gos/go1.4.3/src/net/tcpsock_posix.go:234 +0x4e
net.(*TCPListener).Accept(0xc20802a9f0, 0x0, 0x0, 0x0, 0x0)
	/root/.gvm/gos/go1.4.3/src/net/tcpsock_posix.go:244 +0x4c
net/http.(*Server).Serve(0xc20819c000, 0x7f81bbb81758, 0xc20802a9f0, 0x0, 0x0)
	/root/.gvm/gos/go1.4.3/src/net/http/server.go:1728 +0x92
net/http.Serve(0x7f81bbb81758, 0xc20802a9f0, 0x7f81bbb86750, 0xc20814a070, 0x0, 0x0)
	/root/.gvm/gos/go1.4.3/src/net/http/server.go:1606 +0xa1
github.com/influxdb/influxdb/services/meta.(*Service).serve(0xc2080aca00)
	/tmp/tmp.WXyOm7f2Na/src/github.com/influxdb/influxdb/services/meta/service.go:131 +0x7a
created by github.com/influxdb/influxdb/services/meta.(*Service).Open
	/tmp/tmp.WXyOm7f2Na/src/github.com/influxdb/influxdb/services/meta/service.go:123 +0xa06

goroutine 16 [select]:
github.com/influxdb/influxdb/cmd/influxd/run.(*Server).monitorErrorChan(0xc208082180, 0xc20805a540)
	/tmp/tmp.WXyOm7f2Na/src/github.com/influxdb/influxdb/cmd/influxd/run/server.go:594 +0x14a
created by github.com/influxdb/influxdb/cmd/influxd/run.func·003
	/tmp/tmp.WXyOm7f2Na/src/github.com/influxdb/influxdb/cmd/influxd/run/server.go:388 +0x46d

goroutine 18 [select]:
github.com/influxdb/influxdb/cmd/influxd/run.(*Command).monitorServerErrors(0xc208062480)
	/tmp/tmp.WXyOm7f2Na/src/github.com/influxdb/influxdb/cmd/influxd/run/command.go:143 +0x2dd
created by github.com/influxdb/influxdb/cmd/influxd/run.(*Command).Run
	/tmp/tmp.WXyOm7f2Na/src/github.com/influxdb/influxdb/cmd/influxd/run/command.go:125 +0xf3a
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -35,6 +35,7 @@
 - [#5455](https://github.com/influxdata/influxdb/issues/5455): panic: runtime error: slice bounds out of range when loading corrupted wal segment
 - [#5478](https://github.com/influxdata/influxdb/issues/5478): panic: interface conversion: interface is float64, not int64
 - [#5475](https://github.com/influxdata/influxdb/issues/5475): Ensure appropriate exit code returned for non-interactive use of CLI.
+- [#5479](https://github.com/influxdata/influxdb/issues/5479): Bringing up a node as a meta only node causes panic
 
 ## v0.9.6 [2015-12-09]
 

--- a/cmd/influxd/run/server.go
+++ b/cmd/influxd/run/server.go
@@ -363,112 +363,105 @@ func (s *Server) Err() <-chan error { return s.err }
 
 // Open opens the meta and data store and all services.
 func (s *Server) Open() error {
-	if err := func() error {
-		// Start profiling, if set.
-		startProfile(s.CPUProfile, s.MemProfile)
+	// Start profiling, if set.
+	startProfile(s.CPUProfile, s.MemProfile)
 
-		// Open shared TCP connection.
-		ln, err := net.Listen("tcp", s.BindAddress)
-		if err != nil {
-			return fmt.Errorf("listen: %s", err)
+	// Open shared TCP connection.
+	ln, err := net.Listen("tcp", s.BindAddress)
+	if err != nil {
+		return fmt.Errorf("listen: %s", err)
+	}
+	s.Listener = ln
+
+	// Multiplex listener.
+	mux := tcp.NewMux()
+	go mux.Serve(ln)
+
+	if s.MetaService != nil {
+		s.MetaService.RaftListener = mux.Listen(meta.MuxHeader)
+		// Open meta service.
+		if err := s.MetaService.Open(); err != nil {
+			return fmt.Errorf("open meta service: %s", err)
 		}
-		s.Listener = ln
+		go s.monitorErrorChan(s.MetaService.Err())
+	}
 
-		// Multiplex listener.
-		mux := tcp.NewMux()
-		go mux.Serve(ln)
-
-		if s.MetaService != nil {
-			s.MetaService.RaftListener = mux.Listen(meta.MuxHeader)
-
-			// Open meta service.
-			if err := s.MetaService.Open(); err != nil {
-				return fmt.Errorf("open meta service: %s", err)
-			}
-			go s.monitorErrorChan(s.MetaService.Err())
-		}
-
-		if s.TSDBStore != nil {
-			if err := s.initializeDataNode(); err != nil {
-				return err
-			}
-
-			// Append services.
-			s.appendClusterService(s.config.Cluster)
-			s.appendPrecreatorService(s.config.Precreator)
-			s.appendSnapshotterService()
-			s.appendCopierService()
-			s.appendAdminService(s.config.Admin)
-			s.appendContinuousQueryService(s.config.ContinuousQuery)
-			s.appendHTTPDService(s.config.HTTPD)
-			s.appendCollectdService(s.config.Collectd)
-			if err := s.appendOpenTSDBService(s.config.OpenTSDB); err != nil {
-				return err
-			}
-			for _, g := range s.config.UDPs {
-				s.appendUDPService(g)
-			}
-			s.appendRetentionPolicyService(s.config.Retention)
-			for _, g := range s.config.Graphites {
-				if err := s.appendGraphiteService(g); err != nil {
-					return err
-				}
-			}
-
-			s.Subscriber.MetaClient = s.MetaClient
-			s.ShardMapper.MetaClient = s.MetaClient
-			s.QueryExecutor.MetaClient = s.MetaClient
-			s.ShardWriter.MetaClient = s.MetaClient
-			s.HintedHandoff.MetaClient = s.MetaClient
-			s.Subscriber.MetaClient = s.MetaClient
-			s.PointsWriter.MetaClient = s.MetaClient
-			s.Monitor.MetaClient = s.MetaClient
-
-			s.ClusterService.Listener = mux.Listen(cluster.MuxHeader)
-			s.SnapshotterService.Listener = mux.Listen(snapshotter.MuxHeader)
-			s.CopierService.Listener = mux.Listen(copier.MuxHeader)
-
-			// Open TSDB store.
-			if err := s.TSDBStore.Open(); err != nil {
-				return fmt.Errorf("open tsdb store: %s", err)
-			}
-
-			// Open the hinted handoff service
-			if err := s.HintedHandoff.Open(); err != nil {
-				return fmt.Errorf("open hinted handoff: %s", err)
-			}
-
-			// Open the subcriber service
-			if err := s.Subscriber.Open(); err != nil {
-				return fmt.Errorf("open subscriber: %s", err)
-			}
-
-			// Open the points writer service
-			if err := s.PointsWriter.Open(); err != nil {
-				return fmt.Errorf("open points writer: %s", err)
-			}
-
-			// Open the monitor service
-			if err := s.Monitor.Open(); err != nil {
-				return fmt.Errorf("open monitor: %v", err)
-			}
-
-			for _, service := range s.Services {
-				if err := service.Open(); err != nil {
-					return fmt.Errorf("open service: %s", err)
-				}
-			}
-		}
-
-		// Start the reporting service, if not disabled.
-		if !s.reportingDisabled {
-			go s.startServerReporting()
-		}
-
-		return nil
-
-	}(); err != nil {
+	// initialize MetaClient.
+	if err = s.initializeMetaClient(); err != nil {
 		return err
+	}
+
+	if s.TSDBStore != nil {
+		// Append services.
+		s.appendClusterService(s.config.Cluster)
+		s.appendPrecreatorService(s.config.Precreator)
+		s.appendSnapshotterService()
+		s.appendCopierService()
+		s.appendAdminService(s.config.Admin)
+		s.appendContinuousQueryService(s.config.ContinuousQuery)
+		s.appendHTTPDService(s.config.HTTPD)
+		s.appendCollectdService(s.config.Collectd)
+		if err := s.appendOpenTSDBService(s.config.OpenTSDB); err != nil {
+			return err
+		}
+		for _, g := range s.config.UDPs {
+			s.appendUDPService(g)
+		}
+		s.appendRetentionPolicyService(s.config.Retention)
+		for _, g := range s.config.Graphites {
+			if err := s.appendGraphiteService(g); err != nil {
+				return err
+			}
+		}
+
+		s.Subscriber.MetaClient = s.MetaClient
+		s.ShardMapper.MetaClient = s.MetaClient
+		s.QueryExecutor.MetaClient = s.MetaClient
+		s.ShardWriter.MetaClient = s.MetaClient
+		s.HintedHandoff.MetaClient = s.MetaClient
+		s.Subscriber.MetaClient = s.MetaClient
+		s.PointsWriter.MetaClient = s.MetaClient
+		s.Monitor.MetaClient = s.MetaClient
+
+		s.ClusterService.Listener = mux.Listen(cluster.MuxHeader)
+		s.SnapshotterService.Listener = mux.Listen(snapshotter.MuxHeader)
+		s.CopierService.Listener = mux.Listen(copier.MuxHeader)
+
+		// Open TSDB store.
+		if err := s.TSDBStore.Open(); err != nil {
+			return fmt.Errorf("open tsdb store: %s", err)
+		}
+
+		// Open the hinted handoff service
+		if err := s.HintedHandoff.Open(); err != nil {
+			return fmt.Errorf("open hinted handoff: %s", err)
+		}
+
+		// Open the subcriber service
+		if err := s.Subscriber.Open(); err != nil {
+			return fmt.Errorf("open subscriber: %s", err)
+		}
+
+		// Open the points writer service
+		if err := s.PointsWriter.Open(); err != nil {
+			return fmt.Errorf("open points writer: %s", err)
+		}
+
+		// Open the monitor service
+		if err := s.Monitor.Open(); err != nil {
+			return fmt.Errorf("open monitor: %v", err)
+		}
+
+		for _, service := range s.Services {
+			if err := service.Open(); err != nil {
+				return fmt.Errorf("open service: %s", err)
+			}
+		}
+	}
+
+	// Start the reporting service, if not disabled.
+	if !s.reportingDisabled {
+		go s.startServerReporting()
 	}
 
 	return nil
@@ -547,15 +540,19 @@ func (s *Server) reportServer() {
 
 	numMeasurements := 0
 	numSeries := 0
-	for _, di := range dis {
-		d := s.TSDBStore.DatabaseIndex(di.Name)
-		if d == nil {
-			// No data in this store for this database.
-			continue
+
+	// Only needed in the case of a data node
+	if s.TSDBStore != nil {
+		for _, di := range dis {
+			d := s.TSDBStore.DatabaseIndex(di.Name)
+			if d == nil {
+				// No data in this store for this database.
+				continue
+			}
+			m, s := d.MeasurementSeriesCounts()
+			numMeasurements += m
+			numSeries += s
 		}
-		m, s := d.MeasurementSeriesCounts()
-		numMeasurements += m
-		numSeries += s
 	}
 
 	clusterID := s.MetaClient.ClusterID()
@@ -603,8 +600,8 @@ func (s *Server) monitorErrorChan(ch <-chan error) {
 	}
 }
 
-// initializeDataNode will set the MetaClient and join the node to the cluster if needed
-func (s *Server) initializeDataNode() error {
+// initializeMetaClient will set the MetaClient and join the node to the cluster if needed
+func (s *Server) initializeMetaClient() error {
 	// if the node ID is > 0 then we just need to initialize the metaclient
 	if s.Node.ID > 0 {
 		s.MetaClient = meta.NewClient(s.Node.MetaServers, s.metaUseTLS)
@@ -626,17 +623,20 @@ func (s *Server) initializeDataNode() error {
 		}
 		s.MetaClient = meta.NewClient([]string{s.MetaService.HTTPAddr()}, s.metaUseTLS)
 	} else {
-		// join this data node to the cluster
+		// join this node to the cluster
 		s.MetaClient = meta.NewClient(s.joinPeers, s.metaUseTLS)
 	}
 	if err := s.MetaClient.Open(); err != nil {
 		return err
 	}
-	n, err := s.MetaClient.CreateDataNode(s.httpAPIAddr, s.tcpAddr)
-	if err != nil {
-		return err
+
+	if s.TSDBStore != nil {
+		n, err := s.MetaClient.CreateDataNode(s.httpAPIAddr, s.tcpAddr)
+		if err != nil {
+			return err
+		}
+		s.Node.ID = n.ID
 	}
-	s.Node.ID = n.ID
 	metaNodes, err := s.MetaClient.MetaNodes()
 	if err != nil {
 		return err

